Enhancing Delete Confirmation for Record, Workspace and Account Manager

tsuchiyama-araya opened the pull request:
### Content
Enhancing Delete Confirmation for Record, Workspace and Account Manager to prevent unwanted Delete.
Also, DeleteConfirmModal exists therefor, reusing it might be the best practice so I used DeleteConfirmModal to all delete process.

Adding enhancement on backend where you cannot delete workspace and account if there is workflow running.
Also Implemented cannot delete workspace UI when there is running workflow.

### Testcase
https://docs.google.com/spreadsheets/d/1g32sdfAKUDfodGAMxGzYP_yZmx9gN_UyIgKSycZwq8M/edit?gid=0#gid=0

tsuchiyama-araya committed:
Improvement for Confirmation During Delete for Records

tsuchiyama-araya committed:
Enhancing Delete Confirmation for Account Delete

tsuchiyama-araya committed:
Enhancing Delete Confirmation for Workspace Delete

itutu-tienday commented:
@tsuchiyama-araya 
Is the backend processing (data cleaning processing, the main function of #778) planned to be separate from this PR?

tsuchiyama-araya committed:
Delete Workspace Data File Implemented

tsuchiyama-araya committed:
Enhancing Account Manager Delete

tsuchiyama-araya committed:
Cannot Delete Workspace or Account when there is workflow running.

tsuchiyama-araya commented:
> @tsuchiyama-araya Is the backend processing (data cleaning processing, the main function of #778) planned to be separate from this PR?

I Implemented Backend for this feature. Can you check it? Thank you.

tsuchiyama-araya commented:
testの問題を対応します。

tsuchiyama-araya committed:
cannot delete when running with delete me api

tsuchiyama-araya committed:
add canDelete on Workspaces

tsuchiyama-araya committed:
fixing delete experiment test

tsuchiyama-araya committed:
Enhancing delete process

tsuchiyama-araya committed:
remove old confirm modal for workspace

tsuchiyama-araya committed:
fix record confirm delete

itutu-tienday commented on frontend/src/components/common/DeleteConfirmModal.tsx:
```diff
@@ -45,7 +50,7 @@ const DeleteConfirmModal: FC<DeleteConfirmModalProps> = ({
           <Typography style={{ whiteSpace: "pre-wrap" }}>
             {description}
             This operation cannot be undone. To continue, type
```
- [ ] The style of the dialog can be the same as the existing version (title, description, icon).
- [ ] This is a minor point, but "This operation cannot be undone. " is unnecessary.

itutu-tienday commented on frontend/src/components/Workspace/Experiment/ExperimentTable.tsx:
```diff
@@ -277,25 +284,38 @@ const TableImple = memo(function TableImple() {
-                  {experiment
-                    ? `${experiment.name} (${uid})`
-                    : `Unknown (${uid})`}
+        content={(() => {
```
The Record delete dialog can be left as is (the impact of operational mistakes is small).

itutu-tienday commented on studio/app/common/core/workspace/workspace_services.py:
```diff
@@ -120,3 +149,45 @@ def sync_workspace_experiment(cls, db: Session, workspace_id):
             db.bulk_save_objects(exp_records)
+
+    @classmethod
+    def delete_workspace_data(cls, workspace_id: int):
```
It seems that files remain in DIRPATH.INPUT_DIR (files uploaded by users). Cleaning is required.

tsuchiyama-araya committed:
指摘内容修正 Delete Confirm Dialog to have title and Icon

tsuchiyama-araya committed:
Revert Experiment Delete

tsuchiyama-araya committed:
Fix Cancel Button next resetting confirmation delete

tsuchiyama-araya committed:
add transaction with delete workspace and account

tsuchiyama-araya committed:
delete input directory

tsuchiyama-araya committed:
Deleting records when workspace and account is deleted

tsuchiyama-araya committed:
merge develop-main

itutu-tienday commented on studio/app/common/core/workspace/workspace_services.py:
```diff
@@ -120,3 +149,45 @@ def sync_workspace_experiment(cls, db: Session, workspace_id):
+                shutil.rmtree(experiment_path)
+                logger.info(f"Deleted experiment data at: {experiment_path}")
+            except Exception as e:
+                logger.error(f"Failed to delete {experiment_path}: {e}")
```
After the exception is `except`ed here, `shutli.rmtree(workspace_dir)` is executed in the later process, so won't an inconsistent state occur?

- Example of inconsistency
  - The experiment record remains in the DB, but the corresponding workspaces record has disappeared (or it cannot be deleted due to a foreign key error?)

Organize and test the abnormal patterns.

itutu-tienday commented on studio/app/common/core/workspace/workspace_services.py:
```diff
@@ -92,6 +105,22 @@ def delete_workspace_experiment(cls, db: Session, workspace_id, unique_id):
+                ExperimentRecord.workspace_id == workspace_id
+            )
+        )
+        db.commit()
```
Please verify the following:

- https://github.com/arayabrain/barebone-studio/pull/790/files/fb06033052d20eae8be761cdb859e2342ca18bc1#r2086414158

itutu-tienday commented on studio/app/common/routers/workspace.py:
```diff
@@ -220,20 +251,31 @@ def delete_workspace(
+
+        WorkspaceService.delete_workspace_data(workspace_id=workspace_id)
+
+        ws.deleted = True
```
This process should be normalized.
Consolidate it into WorkspaceService.delete_workspace_data

tsuchiyama-araya replied:
I normalized using this kind of implementation.
https://github.com/arayabrain/barebone-studio/pull/790/files#diff-b0f435d216426c85330b91a8f4dc3e176505e461ac61eaeb83720924c57fc911R225

itutu-tienday commented on studio/app/common/routers/users_admin.py:
```diff
@@ -71,6 +77,41 @@ async def delete_user(
+            db, user_id, organization_id=current_admin.organization.id
+        )
+
+        db.commit()
```
Although the commit is done here, the transaction management is a little loose.

- For example, if the subsequent `WorkspaceService.delete_workspace_data` is interrupted due to some exception, a file that can never be deleted will remain.

Refactor the transaction to eliminate the possibility that a physical file will remain in an undeletable state.

#### Supplemental
- It seems appropriate to store `WorkspaceService.delete_workspace_experiment_by_workspace_id` and `WorkspaceService.delete_workspace_data_by_user_id` in `WorkspaceService.delete_workspace_data`. (The transaction is controlled within that.)
- It seems good to separate the commit into two stages, for deleting the workspace and deleting the user record. (Since there is little data dependency between the two.)

tsuchiyama-araya replied:
I think this is implemented with this updated code.
https://github.com/arayabrain/barebone-studio/pull/790/commits/a047e5adfc8599c8814507bf27172ba10becce51

itutu-tienday commented on frontend/src/components/common/DeleteConfirmModal.tsx:
```diff
@@ -59,7 +87,7 @@ const DeleteConfirmModal: FC<DeleteConfirmModalProps> = ({
+          <Button onClick={onCancel} variant={"outlined"}>
             CANCEL
           </Button>
           <Button onClick={onConfirm} color={"error"} variant="contained">
```
If the confirmation input doesn't match, the button should be disabled (as it was previously implemented).

tsuchiyama-araya committed:
fixing cannot delete with experiement after conflict fix

tsuchiyama-araya commented on frontend/src/components/Workspace/Experiment/ExperimentTable.tsx:
```diff
@@ -277,25 +284,38 @@ const TableImple = memo(function TableImple() {
+            .filter(Boolean) // removes undefined/null
+            .join(", ")
+
+          return (
```
Fixed this. Yes it seems it happened after fixing conflicts.

tsuchiyama-araya committed:
Delete Confirm Button UI & Delete Database first before Deleting direct files or user

tsuchiyama-araya committed:
Merge branch 'develop-main' into feature/delete_enhancement

tsuchiyama-araya committed:
fixing test

tsuchiyama-araya committed:
add warning icon with account manager delete

tsuchiyama-araya committed:
fix delete workspace

tsuchiyama-araya committed:
指摘修正

itutu-tienday commented on studio/app/common/core/workspace/workspace_services.py:
```diff
@@ -131,3 +144,74 @@ def sync_workspace_experiment(cls, db: Session, workspace_id: str):
+        workspace_dir = join_filepath([DIRPATH.OUTPUT_DIR, str(workspace_id)])
+        if os.path.exists(workspace_dir):
+            for experiment_id in os.listdir(workspace_dir):
+                if ExperimentRecord.exists(workspace_id, uid=str(experiment_id)):
```
What is this check for? (Please explain why it is necessary.)

tsuchiyama-araya replied:
It seems like my previous implementation is the reason why its logging.
So when i checked it after there is no need for this check.

I removed it in the next commit

itutu-tienday commented on studio/app/common/routers/users_admin.py:
```diff
@@ -71,6 +77,87 @@ async def delete_user(
+                db.add(ws)
+                db.commit()
+
+                WorkspaceService.delete_workspace_experiment_files(
```
Fundamentally, the following points seem to have not yet been resolved:

https://github.com/arayabrain/barebone-studio/pull/790/files/fb06033052d20eae8be761cdb859e2342ca18bc1#r2084173421

*For example, if an exception occurs during file deletion (delete_workspace_experiment_files), won't junk files remain because the db record has been lost?

tsuchiyama-araya replied:
This is fixed with updated code.
You can also check this commit.

https://github.com/arayabrain/barebone-studio/pull/790/commits/a047e5adfc8599c8814507bf27172ba10becce51

itutu-tienday commented on studio/app/common/core/workspace/workspace_services.py:
```diff
@@ -131,3 +144,74 @@ def sync_workspace_experiment(cls, db: Session, workspace_id: str):
+    def delete_workspace_files(cls, directory: str):
+        try:
+            if os.path.exists(directory):
+                shutil.rmtree(directory)
```
Dangerous because it rmtrees an externally specified path. (Similar to rm -fr)

Pass the required ID as an argument instead of the path, and build the path within the function.

itutu-tienday replied:
> Pass the required ID as an argument instead of the path, and build the path within the function.

It seems that the above is not implemented yet.

tsuchiyama-araya replied:
Fixed This

itutu-tienday commented on studio/app/common/core/workspace/workspace_services.py:
```diff
@@ -131,3 +144,74 @@ def sync_workspace_experiment(cls, db: Session, workspace_id: str):
+                    f"Skipping deletion of running experiment '{experiment_id}'"
+                )
+
+            shutil.rmtree(experiment_path)
```
Dangerous because it rmtrees an externally specified path. (Similar to rm -fr)

Pass the required ID as an argument instead of the path, and build the path within the function.

itutu-tienday replied:
Confirmed that the usage has been switched to `ExptDataWriter.delete_data`.

itutu-tienday commented on studio/app/common/core/workspace/workspace_services.py:
```diff
@@ -131,3 +144,74 @@ def sync_workspace_experiment(cls, db: Session, workspace_id: str):
+                    )
+
+    @classmethod
+    def delete_experiement_files(cls, experiment_path: str, experiment_id: str):
```
Why was this function added separately from the existing record deletion process? (Isn't the function redundant?)

itutu-tienday replied:
Confirmed that the usage has been switched to `ExptDataWriter.delete_data`.

tsuchiyama-araya committed:
process_workspace_deletion create and refactoring

tsuchiyama-araya commented:
@itutu-tienday 
I updated the testcase.
https://docs.google.com/spreadsheets/d/1g32sdfAKUDfodGAMxGzYP_yZmx9gN_UyIgKSycZwq8M/edit?gid=0#gid=0

Also I fixed the review points.
Please Check it again. Thank you

itutu-tienday commented on studio/app/common/core/utils/file_reader.py:
```diff
@@ -97,6 +98,20 @@ def parse(self, content: bytes) -> dict:
         return {"raw": content}
 
 
+class ExperimentReader:
```
There seems to be no need to separate it into a separate class from ExptConfigReader (is this intentional? Or is this due to a lack of understanding of the existing library specifications?)

tsuchiyama-araya replied:
Im sorry. While I was fixing the code. I had to add function to ExperimentWriter file and called it into workspace_services. 
But I bumped to problem that if the 2 files is importing each other it would cause circular import error. That's why I changed it into different file.

I can changed it into lazy importing to import it into the function its used.

itutu-tienday replied:
Can't the target function simply be defined in ExptConfigReader?
(Even from a semantic point of view, storing it in ExptConfigReader seems appropriate.)

itutu-tienday replied:
@tsuchiyama-araya 
> (Even from a semantic point of view, storing it in ExptConfigReader seems appropriate.)

Could you please address the above? (If you cannot, please tell me the reason.)

tsuchiyama-araya replied:
Im sorry, I miss the part of you stating it to store in ExptConfigReader.
I will add this to the next commit. Thank you.

tsuchiyama-araya committed:
returning function to workspace service

itutu-tienday commented on studio/app/common/core/experiment/experiment_writer.py:
```diff
@@ -128,12 +131,31 @@ def __init__(
+            yaml_path = join_filepath([experiment_path, "experiment.yaml"])
+            status = ExperimentReader.load_experiment_success_status(yaml_path)
+
+            if status == "running":
```
When referencing workflow status, use enum constants instead of hard-coded string values.

Constants are provided below:
- #539
  - https://github.com/arayabrain/barebone-studio/pull/539/files#diff-6eac29ff5fd1d7ccd7c583afb4badcff84597d5ea05cbfef183b20c0ab55e994R11

tsuchiyama-araya replied:
FIXED

itutu-tienday commented on studio/app/common/routers/workspace.py:
```diff
@@ -56,10 +64,31 @@ def search_workspaces(
+                        continue
+                    yaml_path = join_filepath([experiment_path, "experiment.yaml"])
+                    status = ExperimentReader.load_experiment_success_status(yaml_path)
+                    if status == "running":
```
When referencing workflow status, use enum constants instead of hard-coded string values.

Constants are provided below:
- #539
  - https://github.com/arayabrain/barebone-studio/pull/539/files#diff-6eac29ff5fd1d7ccd7c583afb4badcff84597d5ea05cbfef183b20c0ab55e994R11

tsuchiyama-araya replied:
FIXED

tsuchiyama-araya committed:
changing running word into enum

tsuchiyama-araya committed:
Fix delete files directory to not pass directly

itutu-tienday commented on studio/app/common/core/experiment/experiment_writer.py:
```diff
@@ -128,12 +130,35 @@ def __init__(
         )
 
+        try:
+            yaml_path = join_filepath([experiment_path, "experiment.yaml"])
```
For "experiment.yaml", use the DIRPATH class constant.

tsuchiyama-araya replied:
FIX

itutu-tienday commented on studio/app/common/core/experiment/experiment_writer.py:
```diff
@@ -128,12 +130,35 @@ def __init__(
+            status = load_experiment_success_status(yaml_path)
+
+            if status == WorkflowRunStatus.RUNNING.value:
+                logger.warning(
```
After logging, it appears that the program is not skipping any steps and is instead moving directly to the next step (shuutil.rmtree).
Is this okay?

If it is a problem, please fix the test case.

tsuchiyama-araya replied:
FIX

tsuchiyama-araya replied:
This wont move to next step anymore since I am returning False when it fails

itutu-tienday replied:
The above issue seems to have been fixed, but it seems that the following issue has occurred due to the issue.
- https://github.com/arayabrain/barebone-studio/pull/790/files#r2096898169

tsuchiyama-araya replied:
I am adding the fix for this to the next commit.
Thank you.

tsuchiyama-araya committed:
指摘修正

tsuchiyama-araya commented:
@itutu-tienday 
I fixed your review. can you check it again. Im sorry. Thank you.

tsuchiyama-araya commented on studio/app/common/routers/workspace.py:
```diff
@@ -56,10 +67,31 @@ def search_workspaces(
+                    experiment_path = join_filepath([workspace_dir, experiment_id])
+                    if not os.path.isdir(experiment_path):
+                        continue
+                    yaml_path = join_filepath([experiment_path, "experiment.yaml"])
```
FIX

itutu-tienday commented on studio/app/common/core/experiment/experiment_writer.py:
```diff
@@ -128,12 +130,36 @@ def __init__(
+                logger.warning(
+                    f"Skipping deletion of running experiment '{self.unique_id}'"
+                )
+                return False
```
If you do the following, you can delete a workspace even if there is a workflow currently running.

1. Display the same workspaces screen in two browser tabs
2. In the first browser, run a workflow in a workspace (let's call it workspace-1)
3. In the second tab, delete workspace-1 → It can be deleted

tsuchiyama-araya replied:
I fixed this. 
Tested to make sure the workflow cant be delete from other tabs.

itutu-tienday replied:
The above issue seems to have been fixed, but it seems that the following issue has occurred due to the issue.
- https://github.com/arayabrain/barebone-studio/pull/790/files#r2096898169

tsuchiyama-araya replied:
Fixing this right now.

itutu-tienday commented on studio/app/common/core/utils/file_reader.py:
```diff
@@ -97,6 +98,20 @@ def parse(self, content: bytes) -> dict:
+
+        try:
+            with open(yaml_path, "r") as file:
+                data = yaml.safe_load(file)
```
Please access yaml files via `ConfigReader/ConfigWriter` to maintain parallelism and exclusive management.
(Do not use raw yaml.safe_load)

The above refactoring was implemented in the following PR:
- #796

tsuchiyama-araya replied:
I refactor it to read with ConfigReader

itutu-tienday replied:
- [ ] The "unknown" status is used as the default value, but since no such status exists, use None if you are comfortable with it.
- [ ] If the corresponding experiment.yaml does not exist, an exception will be thrown and the workspace list will not be displayed at all.
→ Please handle the error so that the user does not become unable to operate.

tsuchiyama-araya replied:
I will add those reviews on the next commit.
Thank you.

tsuchiyama-araya replied:
THank you.
I fix this code with this commit.

https://github.com/arayabrain/barebone-studio/pull/790/commits/d4368438fbf2b8293898817cafd1e3cea16151b1

itutu-tienday commented on frontend/src/components/Workspace/Experiment/ExperimentTable.tsx:
```diff
@@ -277,8 +284,8 @@ const TableImple = memo(function TableImple() {
         )}
       </Box>
       <ConfirmDialog
```
When multiple deletions are performed using the button ("delete-selected-button"), the deletions cannot be performed.

- An error message "Please type DELETE to confirm" is displayed.
- Please improve the above. Please also specify the above test items in the test case.

tsuchiyama-araya replied:
I Fixed this and tested. All Experiments deleted

tsuchiyama-araya committed:
指摘修正。別タブで削除可能＆Update delete_workspace_storage_files＆Access Yaml Path with ConfigReader

tsuchiyama-araya commented on studio/app/common/core/workspace/workspace_services.py:
```diff
@@ -131,3 +149,96 @@ def sync_workspace_experiment(cls, db: Session, workspace_id: str):
             db.bulk_save_objects(exp_records)
+
+    @classmethod
+    def delete_workspace_experiment_files(
```
Changed name done

itutu-tienday commented on studio/app/common/core/utils/file_reader.py:
```diff
@@ -97,6 +98,20 @@ def parse(self, content: bytes) -> dict:
+                data = yaml.safe_load(file)
+                return data.get("success", "unknown")
+        except yaml.YAMLError:
+            return "error"
```
For workspace status, use a constant instead of hard-coding a string. (Repeat point)

tsuchiyama-araya replied:
Added this with the new commit

itutu-tienday commented on studio/app/common/core/workspace/workspace_services.py:
```diff
@@ -131,3 +149,96 @@ def sync_workspace_experiment(cls, db: Session, workspace_id: str):
+    def process_workspace_deletion(cls, db: Session, workspace_id: str, user_id: str):
+        try:
+            # Delete experiment from database
+            WorkspaceService.delete_experiment_records_by_workspace_id(db, workspace_id)
```
@tsuchiyama-araya 
Has this been addressed here?

tsuchiyama-araya replied:
I already added the code after the `if not ws`

itutu-tienday commented on studio/app/common/core/experiment/experiment_writer.py:
```diff
@@ -128,12 +130,35 @@ def __init__(
+                logger.warning(
+                    f"Skipping deletion of running experiment '{self.unique_id}'"
+                )
+                return False
```
A workspace with 0 records cannot be deleted.
The following error occurred.

```
2025-05-20 13:33:57,592 ERROR:    [optinist] (69543) process_workspace_deletion():242 - Error deleting or updating workspace 23: local variable 'hasDeleteDataArr' referenced before assignment
Traceback (most recent call last):
  File "/workspaces/barebone-studio/studio/app/common/core/workspace/workspace_services.py", line 233, in process_workspace_deletion
    cls.delete_workspace_storage_files(workspace_id=workspace_id)
  File "/workspaces/barebone-studio/studio/app/common/core/workspace/workspace_services.py", line 171, in delete_workspace_storage_files
    if all(hasDeleteDataArr):
UnboundLocalError: local variable 'hasDeleteDataArr' referenced before assignment
```

tsuchiyama-araya replied:
I added this with the new commit and fix

tsuchiyama-araya committed:
指摘修正&workspace with 0 to be able to delete & add load yaml to exptConfig Reader

tsuchiyama-araya committed:
Fix Test and Refactor Workflow Yaml Status

itutu-tienday commented on studio/tests/app/common/routers/test_experiment.py:
```diff
@@ -29,6 +29,11 @@ def test_delete(client):
     os.makedirs(dirpath, exist_ok=True)
+
+    # Add dummy experiment.yaml
+    with open(os.path.join(dirpath, "experiment.yaml"), "w") as f:
```
A minor point, but please use constants in "experiment.yaml".

- Previously pointed out
  - https://github.com/arayabrain/barebone-studio/pull/790/files/149094b0774ea0960c2789583f31e06f132d6352#r2094990567

tsuchiyama-araya replied:
Fixing this into next commit.

itutu-tienday commented on studio/app/common/routers/workspace.py:
```diff
@@ -56,10 +65,32 @@ def search_workspaces(
+                    experiment_path = join_filepath([workspace_dir, experiment_id])
+                    if not os.path.isdir(experiment_path):
+                        continue
+                    status = ExptConfigReader.load_experiment_success_status(
```
If load_experiment_success_status returns None, an exception is raised.

Also, it seems that None is not taken into account in the subsequent status determination. (What kind of behavior will this result in?)

tsuchiyama-araya replied:
If it returns none when deleting it will not delete the experiment. Also since it returns false it will not delete the workflow storage files.

On read when it returns None the workspace can be deleted. It will not disable the delete button

itutu-tienday replied:
- [ ] Apply the fix as below:
  - https://github.com/arayabrain/barebone-studio/pull/790/files/a047e5adfc8599c8814507bf27172ba10becce51#r2097259429
- [ ] Also, in the link below, None is checked.
  If None is acceptable here, please clarify in the comments.
  - https://github.com/arayabrain/barebone-studio/commit/d4368438fbf2b8293898817cafd1e3cea16151b1#diff-0a19e403500a1e1806fc4a6e72d17053e42e263f7fd916a228ea51f3c437a1faR142

tsuchiyama-araya replied:
Applied this to next commit. Please check it if there is no problem.
Thank you

tsuchiyama-araya replied:
https://github.com/arayabrain/barebone-studio/pull/790/commits/f25fcc8a0229f7e4564dc8547bb089153438dcb6

itutu-tienday replied:
> If None is acceptable here, please clarify in the comments.

The intention should be stated in the "None not checked" part. (None should always be checked. Why is it being skipped?)

itutu-tienday replied:
The fix was confirmed

tsuchiyama-araya committed:
refactor test experiment

itutu-tienday commented on studio/app/common/core/workspace/workspace_services.py:
```diff
@@ -131,3 +136,107 @@ def sync_workspace_experiment(cls, db: Session, workspace_id: str):
+        workspace_dir = join_filepath([DIRPATH.OUTPUT_DIR, str(workspace_id)])
+        if os.path.exists(workspace_dir):
+            for experiment_id in os.listdir(workspace_dir):
+                WorkspaceService.delete_workspace_experiment(
```
Here, commit is executed every time within `delete_workspace_experiment`, so transaction management is broken.

I have a suggestion. Rather than deleting each experiment_records individually, how about deleting them all at once using SQL? (Does it need to be done in a loop?)
Then, how about committing at the caller?

tsuchiyama-araya replied:
https://github.com/arayabrain/barebone-studio/pull/790/commits/f25fcc8a0229f7e4564dc8547bb089153438dcb6

itutu-tienday replied:
A few more improvements have been applied in the following PR:
- #812

itutu-tienday commented on studio/app/common/core/workspace/workspace_services.py:
```diff
@@ -131,3 +136,107 @@ def sync_workspace_experiment(cls, db: Session, workspace_id: str):
+    @classmethod
+    def process_workspace_deletion(cls, db: Session, workspace_id: str, user_id: str):
+        try:
+            # Soft Delete Workspace
```
The comments are inaccurate.

- "# Soft Delete Workspace" -> "# Search for Workspace" etc.
- "cls.delete_workspace_storage_files" -> "# Delete the files in the Workspace" etc.

tsuchiyama-araya replied:
https://github.com/arayabrain/barebone-studio/pull/790/commits/f25fcc8a0229f7e4564dc8547bb089153438dcb6

itutu-tienday replied:
> - "cls.delete_workspace_storage_files" -> "# Delete the files in the Workspace" etc.

The above doesn't seem to have been fixed.

itutu-tienday replied:
The fix was confirmed

tsuchiyama-araya committed:
指摘修正2025/05/21

tsuchiyama-araya committed:
add comments

itutu-tienday commented on studio/app/common/core/experiment/experiment_writer.py:
```diff
@@ -137,9 +137,9 @@ def delete_data(self) -> bool:
+            )
 
-            if status == WorkflowRunStatus.RUNNING.value or status is None:
+            if status == WorkflowRunStatus.RUNNING or status is None:
```
This may be a minor point, but when making conditional decisions that include None, make sure to test for None first.
(This is less likely to cause problems.)

tsuchiyama-araya replied:
Noneを先にチェックすること

itutu-tienday replied:
The fix was confirmed

tsuchiyama-araya committed:
指摘修正

tsuchiyama-araya committed:
fix if else for status

itutu-tienday commented on studio/app/common/routers/workspace.py:
```diff
@@ -56,10 +65,34 @@ def search_workspaces(
+                    )
+                    if status is None:
+                        pass
+                    if status == WorkflowRunStatus.RUNNING:
```
Why is the logic different from the following (elif)?
https://github.com/arayabrain/barebone-studio/pull/790/files#diff-0a19e403500a1e1806fc4a6e72d17053e42e263f7fd916a228ea51f3c437a1faR146

*Make sure to test.

tsuchiyama-araya committed:
fix if else with search workspace yaml status

Commit:
Refactoring WorkspaceService
- Consolidating experiment data deletion into delete_workspace_experiment
- Consolidating workspaces data deletion into def delete_workspace_contents

itutu-tienday committed:
Merge pull request #812 from arayabrain/feature/delete_enhancement_refactor

Refactor feature/delete_enhancement

tsuchiyama-araya commented:
もう一度動作確認しました。
特に問題はなかったです。そのため、マージします。
